src/main/kotlin/com/sourcegraph/cody/history/HistoryService.kt

```diff
@@ -33,10 +33,14 @@ class HistoryService(private val project: Project) :
   }
 
   @Synchronized
-  fun updateChatMessages(internalId: String, chatMessages: List<ChatMessage>) {
+  fun updateChatMessages(
+      internalId: String,
+      chatMessages: List<ChatMessage>,
+      shouldUpdateTime: Boolean
+  ) {
     val found = getOrCreateChat(internalId)
     found.messages = chatMessages.map(::convertToMessageState).toMutableList()
-    if (chatMessages.lastOrNull()?.speaker == Speaker.HUMAN) {
+    if (shouldUpdateTime) {
```

Comment:
I get all that changes, and they are correct, but I do not find that variable passing particularly clean.
I was thinking if we can simplify it somehow and I get one idea:
Shouldn't we update only, if `found.messages` count increased compared to what already is in the history?

Reply:
oh, sounds good!